ros2topic/ros2topic/verb/pub.py

```diff
@@ -27,13 +34,85 @@
 from ros2topic.api import TopicNameCompleter
 from ros2topic.api import TopicTypeCompleter
 from ros2topic.verb import VerbExtension
-from rosidl_runtime_py import set_message_fields
+from rosidl_parser.definition import AbstractNestedType
+from rosidl_parser.definition import NamespacedType
+from rosidl_runtime_py.convert import get_message_slot_types
+from rosidl_runtime_py.import_message import import_message_from_namespaced_type
 from rosidl_runtime_py.utilities import get_message
 import yaml
 
 MsgType = TypeVar('MsgType')
 
 
+def set_message_fields_expanded(node: Node, msg: Any, values: Dict[str, str]) -> List[Any]:
+    """
+    Set the fields of a ROS message.
+
+    This function is copied from :py:func:`rosidl_runtime_py.set_message_fields`, but with a
+    'Node' parameter, required for obtaining the current time.
+    :param msg: The ROS message to populate.
+    :param values: The values to set in the ROS message. The keys of the dictionary represent
+        fields of the message.
+        If 'auto' is passed as a value to a 'std_msgs.msg.Header' field, an empty Header will be
+        instantiated with its 'stamp' field set to the current time.
+        If 'now' is passed as a value to a 'builtin_interfaces.msg.Time' field, its value will be
+        expanded to the current time.
+    :raises AttributeError: If the message does not have a field provided in the input dictionary.
+    :raises TypeError: If a message value does not match its field type.
+    """
+    timestamp_fields = []
+
+    def set_message_fields_expanded_internal(
+            node: Node, msg: Any, values: Dict[str, str],
+            timestamp_fields: List[Any]) -> List[Any]:
+        try:
+            items = values.items()
+        except AttributeError:
+            raise TypeError(
+                "Value '%s' is expected to be a dictionary but is a %s" %
+                (values, type(values).__name__))
+        for field_name, field_value in items:
+            field = getattr(msg, field_name)
+            field_type = type(field)
+            qualified_class_name = '{}.{}'.format(field_type.__module__, field_type.__name__)
+            if field_type is array.array:
+                value = field_type(field.typecode, field_value)
+            elif field_type is numpy.ndarray:
+                value = numpy.array(field_value, dtype=field.dtype)
+            elif type(field_value) is field_type:
+                value = field_value
+            # We can't import these types directly, so we use the qualified class name to
+            # distinguish them from other fields
+            elif qualified_class_name == 'std_msgs.msg._header.Header' and field_value == 'auto':
+                stamp_now = node.get_clock().now().to_msg()
+                value = field_type(stamp=stamp_now, frame_id='')
+                timestamp_fields.append(partial(setattr, value, 'stamp'))
```

Comment:
can we add `continue` here, since `partial`ed function object will be called and overwrite this feild?

```suggestion
                timestamp_fields.append(partial(setattr, value, 'stamp'))
                continue
```

Reply:
Done in bdd896bdc89c46b86b397ff3d8d3730479a7e39d